app/models/user.rb

```diff
@@ -0,0 +1,30 @@
+class User < ActiveRecord::Base
+
+  has_many :pictures
+  has_many :comments
+  has_many :likes
+  has_many :liked_pictures, through: :likes, source: :pictures
+  has_many :commented_pictures, through: :comments, source: :pictures
+
+  validates :user_name, presence: true
+
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, :omniauth_providers => [:facebook]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
+    end
+  end
+
+  def self.new_with_session(params, session)
+    super.tap do |user|
+      if data = session["devise.facebook_data"] && session["devise.facebook_data"]["extra"]["raw_info"]
```

Comment:
Lint/AssignmentInCondition: Assignment in condition - you probably meant to use `==`. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)<br>Metrics/LineLength: Line is too long. [103/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)